runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkContextFactory.java

```diff
@@ -36,7 +36,8 @@ public final class SparkContextFactory {
    * context will be reused for beam pipelines. This property should only be enabled for tests.
    *
    * @deprecated Please use {@link SparkContextOptions} instead to allow for proper lifecycle
```

Comment:
IMHO I think we should guide the user more by saying something like:
"Please use SparkContextRule to allow for automatic control of the SparkContext lifecycle (with a default of reusing it) or directly provide a SparkContext by using SparkContextFactory#setProvidedSparkContext and enabling SparkContextOptions#usesProvidedSparkContext"

Reply:
I've added more guidelines, though `SparkContextRule` only exists in the test scope and is not published ... should somebody try and use the SparkRunner for their own tests instead of the direct runner (for whatever reason) that's might be confusing.

Reply:
when I said user, I meant user of the library in other words a beam dev.

Reply:
your wording in last commit it perfect, much clearer